### ISSUE TITLE
Allow Virtual Machines to be Grouped outside of Clusters

### ISSUE THREAD

**Issue by KjellWolf** [netbox-community/netbox#14915]
### NetBox version

v3.7.0

### Feature type

Data model extension

### Proposed functionality

We would like to be able to group virtual machines, outside of a cluster definition. 
We create "projects" ourselves or for our tenants, as several projects can originate from one tenant, as well as other contacts.

The suggestion would therefore be to create a "VM Groups" section under "Virtualisation > Virtual Machines".
options: 
Main:
 - Name, Status, Description, Tenant

Resources:
- Virtual Machines [Name, Status, Primary IPv4/v6, Platform, Service Names, 

- Contacts and Journal

every other customisation we need is handled with the customisation options already provided 

### Use case

As mentioned, there are often projects that require a logical combination of VMs in one view, but do not form a cluster as such. 

This simplifies the view of which servers belong together to the corresponding customer order and who is the contact person at the customer or internally for the project.

**Comment by jeremystretch** [comment]
It sounds like tags or a custom field would work well for this purpose.

**Comment by KjellWolf** [comment]
Tags and custom flields do not have the Options we need all at once. 
We need this as seperate pages to sort in other custom flields we already use Like Docs. 

It wouldnt give us the clarity of all information we need at once

**Comment by jeremystretch** [comment]
Sorry, what specific functionality are you looking for that wouldn't be possible?

**Comment by stavr666** [comment]
Cluster Type / Cluster Group / Cluster works (almost) perfectly for similar cases:

![image](https://github.com/netbox-community/netbox/assets/84839985/5b135b3e-72e1-48a5-955d-d2d06d89d32e)

Yes, the "cluster" terminology looks confusing for actual projects/tenants/orgs/groups in clouds or containerized deployments (and in some cases, there is missing actual infrastructure context, like assigned VMs for deployments, or multisite structure), but it's still usable for described scenario.

**Comment by KjellWolf** [comment]
@stavr666 yes, I mentioned it. But it's then it's like you said confusing and not every time applicable. 
more so with staff in training. 

I still pitched this as an Idea to my management.

**Comment by uppsju** [comment]
Would an alternative be to use Tenant Groups? E.g for each client you have, you create a separate tenant group. Then you could create one tenant for each project for that customer.

**Comment by jeffgdotorg** [comment]
@KjellWolf I'm triaging some backlogged issues and came across this one. Did @uppsju's suggestion of fielding Tenant Groups to achieve your aim pan out? If so, please close the issue. If not, please make revisions to your original issue body addressing the shortcomings of Tags, Custom Fields, and Tenant Groups.

**Comment by arthanson** [comment]
@KjellWolf any feedback on the above suggestion from uppsju?  I will close this soon if no feedback.

**Comment by KjellWolf** [comment]
While Tenant Groups could be an option, they may not perfectly fit your Client > Project > VM structure, especially with multiple clusters and loose VM groups. The proposed "VM Groups" section offers a more tailored solution. It allows for granular organization, independent of clusters, reflecting your project hierarchy effectively. This ensures seamless management and clarity in customer orders and internal coordination.

**Comment by doc-sheet** [comment]
It would be nice to set resource values for group to share memory / cpu etc limits for all vms in group.

**Comment by jeremystretch** [comment]
After discussing this with the other maintainers, we've opted not to move forward with the proposed functionality. The FR is ambiguous as to the specific functionality to be implemented, and it hasn't received any significant interest from the community.
